snapshot-file-test/src/main/java/com/hazelcast/jet/tests/snapshot/file/GeneratedFilesVerifier.java

```diff
@@ -0,0 +1,203 @@
+import java.util.concurrent.locks.LockSupport;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class GeneratedFilesVerifier extends Thread {
+
+    private static final String FILE_SINK_DIR_FOR_TEST_PATH = "/tmp/file_sink";
+
+    private static final int SLEEP_AFTER_READ_FILES_MS = 1000;
+    private static final int ALLOWED_NO_INPUT_MS = 600000;
+
+    private final Path filesinkDirectory;
+    private final Verifier verifier;
+    private final String name;
+    private final ILogger logger;
+
+    private volatile boolean finished;
+    private volatile Throwable error;
+    private long lastInputTime;
+
+    public GeneratedFilesVerifier(JetInstance client, String name, ILogger logger) {
+        this.name = name;
+        this.logger = logger;
+        verifier = new Verifier(name, logger);
+        filesinkDirectory = Paths.get(FILE_SINK_DIR_FOR_TEST_PATH + name);
+    }
+
+    @Override
+    public void run() {
+        lastInputTime = System.currentTimeMillis();
+        while (!finished) {
+            try {
+                List<Long> processFiles = processFiles();
+                long now = System.currentTimeMillis();
+                if (processFiles.isEmpty()) {
+                    if (now - lastInputTime > ALLOWED_NO_INPUT_MS) {
+                        throw new AssertionError(
+                                String.format("[%s] No new data was added during last %s", name, ALLOWED_NO_INPUT_MS));
+                    }
+                } else {
+                    lastInputTime = now;
+                    verifier.sendToQueue(processFiles);
+                }
+                Thread.sleep(SLEEP_AFTER_READ_FILES_MS);
+            } catch (Exception e) {
+                logger.severe("[" + name + "] Exception thrown during processing files.", e);
+                error = e;
+                finish();
+            }
+        }
+    }
+
+    private List<Long> processFiles() throws IOException {
+        try (Stream<Path> list = Files.list(filesinkDirectory)) {
+            List<Path> collect = list.collect(Collectors.toList());
+            List<String> content = new ArrayList<>();
+            for (Path path : collect) {
+                List<String> fileContent = Files.readAllLines(path);
+                content.addAll(fileContent);
+                Files.deleteIfExists(path);
+            }
+            return content.stream().map(Long::parseLong).collect(Collectors.toList());
+        }
+    }
+
+    public void finish() {
+        verifier.stop();
+        finished = true;
+    }
+
+    public void checkStatus() {
+        if (error != null) {
+            throw new RuntimeException(error);
+        }
+        if (finished) {
+            throw new RuntimeException("[" + name + "] GeneratedFilesVerifier is not running");
+        }
+        verifier.checkStatus();
+    }
+
+    public static class Verifier {
```

Comment:
The logic between main thread and `Verifier` thread looks a bit complex and not optimal
- main thread reads the files and puts the lines to the `Verifier` queue
- Verifier thread then drains this queue to a `PriorityQueue`
- actual verification done on this `PriorityQueue` but if the item arrives late above process repeated

We can merge these 2 threads
- main thread reads lines and adds them to a `PriorityQueue`
- main thread then tries to verify this queue for the next step
- if everything is in order, the queue is cleared at the end of the verification
- if not, we break and main thread reads more lines and adds to the queue again

So a single thread can read and then verify them, this way we eliminate the second thread and the concurrent queue between the threads .